honeycombio/provider.go

```diff
@@ -73,11 +67,33 @@ func Provider() *schema.Provider {
 	}
 
 	provider.ConfigureContextFunc = func(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+		apiKey := os.Getenv("HONEYCOMB_API_KEY")
```

Comment:
In order to match the Framework server's view of configuration, these environment based configs needed to be brought out of the Schema.